files/common/usr/bin/get-hotfix-version

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+#
+# Copyright 2020 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -o errexit
+set -o pipefail
+
+function usage() {
+	echo "Usage: $(basename "$0")"
+	echo
+	echo "Display the hotfix version information for the appliance."
+	exit 2
+}
+
+function hotfix_version() {
+  local output
+	output=$(zfs get -Hpo value "com.delphix:hotfix-version" "$(dirname "$(zfs list -Hpo name /)")" | tr -d '\n')
+	if [[ "$output" != "-" ]]; then
+	   echo "$output"
+	fi
```

Comment:
Not sure if you think this would be "too clever" and make the code more difficult to understand, but we could accomplish by adding `sed 's/^-$//'` to the pipeline.. E.g.
```
#
# When the "hotfix-version" property doesn't exist (e.g. when no hotfix has been applied), we want to return the
# empty string. We accomplish this by replacing the "-" string returned by "zfs get" with the empty string.
#
zfs get -Hpo value "com.delphix:hotfix-version" "$(dirname "$(zfs list -Hpo name /)")" | sed 's/^-$//' | tr -d '\n'
```

The comment helps make it clear what we're doing, and then we have one less variable and conditional; which makes it more simple, IMO.

I'm good with what you have, but just wanted to offer a potential alternative solution.